python/tests/test_config.py

```diff
@@ -57,6 +61,47 @@ def test_get_population_properties(self):
         self.assertEqual(edge_prop.types_path, '')
         self.assertTrue(edge_prop.elements_path.endswith('tests/data/edges1.h5'))
 
+    def test_partial(self):
+        contents = { "metadata": { "type": "NOT A TYPE" }, }
+        self.assertRaises(SonataError, CircuitConfig, json.dumps(contents), PATH)
+
+        contents = { "metadata": { "type": "partial" }, }
+        cc = CircuitConfig(json.dumps(contents), PATH)
+        self.assertEqual(cc.type, CircuitConfigType.Partial)
+        self.assertEqual(cc.node_populations, set())
+        self.assertEqual(cc.edge_populations, set())
+
+        contents = {
+            "metadata": { "type": "partial" },
+            "networks-mispelled": { },
+            }
+        cc = CircuitConfig(json.dumps(contents), PATH)
+        self.assertEqual(cc.type, CircuitConfigType.Partial)
+        self.assertEqual(cc.node_populations, set())
+        self.assertEqual(cc.edge_populations, set())
+
+        contents = {
+            "metadata": { "type": "partial" },
```

Comment:
partial is a property that is defined on a per population basis because we can have a mix of complete circuits with partial populations in various building steps, especially when the ngv is considered.

We can also have it at the root level so that libsonata treats it with extra care, however we also need the per population partial definition so that we know which population is complete and which isn't.

For example:
```json
{
  "version": 2,
  "manifest": {
    "$BASE_DIR": "."
  },
  "networks": {
    "nodes": [
      {
        "nodes_file": "...",
        "populations": {
          "root_neurons": {
            "type": "biophysical",
            "partial": [
              "cell-properties",
              "morphologies",
              "emodels"
            ],
            "morphologies_dir": "...",
            "biophysical_neuron_models_dir": "..."
          }
        }
      }
    ]
  }

```

Reply:
That's a good point; however, I think for the initial version, we can make the whole thing partial, and then work towards making individual parts partial.

Reply:
Sure, that's fine with me!